### PR TITLE
Add version number based on git tag and -V to print version number. 

### PR DESCRIPTION
**Pull request by weetmuts**
The makefile code will use the git history to create a VERSION string and a COMMIT string.
The VERSION string looks like:

1.0.1
if you have built a clean clone where the HEAD is tagged with "1.0.1"
(you do: "git tag 1.0.1" to tag the current HEAD, then "git push --tags" to push the tags to github.)

1.0.1 with local changes
if you have built 1.0.1 with un-committed changes in your repo.

1.0.1-1-gc2897ad
if you have built 1.0.1 with one committed change after the tag.

1.0.1-1-gc2897ad with local changes
if you have built 1.0.1 with one committed change and you have un-committed changes.

1.0.1-2-gc1138ec
if you have built 1.0.1 with two committed changes after the tag.

speed_1.0.1-2-gc1138ec
if you have built the development branch speed with two committed changes after the tag.

---------------------------

rtl_wmbus -V 
will print for example:

rtl_wmbus: speed_1.0.1-2-gc1138ec
c1138ece0aebecc9c4304dda958591d9cae4673c

### CODE REVIEW

File: rtl_wmbus.c [hunk 1]
```diff
@@ -848,6 +855,9 @@ static void process_options(int argc, char *argv[])
         case 'v':
             opts_show_used_algorithm = 1;
             break;
+        case 'V':
+            print_version();
```
Comment (dwrobel) [inline rtl_wmbus.c:860]: exit(EXIT_SUCCESS); is missing here.

File: Makefile [hunk 2]
```diff
@@ -10,15 +10,54 @@ CFLAGS_WARNINGS?=-Wall -W -Waggregate-return -Wbad-function-cast -Wcast-align -W
...
+TAG_COMMIT_HASH?=$(shell git show-ref --tags | grep $(TAG) | cut -f 1 -d ' ')
 
-release: $(OUTDIR)
+ifeq ($(BRANCH),master)
+  BRANCH:=
+else
+  BRANCH:=$(BRANCH)_
+endif
+
+ifeq ($(COMMIT),$(TAG_COMMIT))
```
Comment (dwrobel) [inline Makefile]: Neither `COMMIT` nor `TAG_COMMIT` was used previously. Did you mean `COMMIT_HASH` and `TAG_COMMIT_HASH` respectively?

File: rtl_wmbus.c (hunk 1 above)
Comment (weetmuts) [reply rtl_wmbus.c:860]: But, it reports exit(EXIT_SUCCESS) below? Do you mean that it is a standard to report failure when printing the version number successfully?

File: Makefile (hunk 2 above)
Comment (weetmuts) [reply Makefile]: So, I copied some code from my wmbusmeters makefile, then I discovered that it had some bugs.
So I removed it. Git itself generates a nice tag string, no need for checking if it is on a tagged commit.

File: rtl_wmbus.c (hunk 1 above)
Comment (dwrobel) [reply rtl_wmbus.c:860]: I mean it should exit with success, now if you run it with `-V` option it gets stuck.

Comment (weetmuts) [reply rtl_wmbus.c:860]: Ah, but are you looking at the earlier version of the pull-request?
The latest version does have the exit. I can repush with force if you like.

Comment (dwrobel) [reply rtl_wmbus.c:860]: I see it now. No need to force-push.

**Pull request merged**